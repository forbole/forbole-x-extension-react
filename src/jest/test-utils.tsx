import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider } from '@mui/material';
import lightTheme from '../config/theme/lightTheme';
import '../assets/locales/i18n';

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>;
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
