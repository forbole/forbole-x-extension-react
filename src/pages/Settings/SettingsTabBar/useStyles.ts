import {useRecoilState} from "recoil";
import {themeState} from "../../../recoil/general";
import { makeStyles} from '@material-ui/styles'

const useStyles = () => {
    const [theme] = useRecoilState(themeState)

    // hacky way to override mui colors
    const styleOverride: { [index: string]: any } = {
        light: {
            indicator: {
                backgroundColor: '#5C7BFF',
                color: '#5C7BFF'
            }
        },
        dark: {
            indicator: {
                backgroundColor: '#E6E6E6',
                color: '#E6E6E6'
            }
        },
    }

    return makeStyles(
        () => ({   indicator: {
                ...styleOverride[theme].indicator,
                width: '8px' // mui uses 8px incremental spacing
            },
            tab:{
                '& .MUI-selected': {
                    ...styleOverride[theme].indicator,
                },
            },
            customTab: {
                minWidth: 80
            }})
    )();
};

export default useStyles;