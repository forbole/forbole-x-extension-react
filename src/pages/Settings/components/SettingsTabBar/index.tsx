import React from 'react'
import { Tabs, Tab } from '@mui/material'
import GeneralSettings from '../GeneralSettings'

type Props = {
  // The tabs to be rendered
  tabs: { label: string }[]

  // The current tab
  currentTab: number

  // When the user clicks on a different tab
  onChange: (_value: number) => void
}

const SettingsTabBar = ({ tabs, currentTab, onChange }: Props) => {
  const tabRender = React.useMemo(() => {
    return tabs.map((tab) => (
      <Tab
        sx={{
          minWidth: 80,
        }}
        key={tab.label}
        label={tab.label}
      />
    ))
  }, [currentTab])

  const tabContent = React.useMemo(
    () => ({
      0: <GeneralSettings />,
    }),
    []
  )

  return (
    <>
      <Tabs
        sx={{
          indicator: {
            color: 'white',
            width: '8px',
          },
        }}
        value={currentTab}
        onChange={(e, v) => {
          onChange(v)
        }}
      >
        {tabRender}
      </Tabs>

      {tabContent[currentTab]}
    </>
  )
}

export default SettingsTabBar
