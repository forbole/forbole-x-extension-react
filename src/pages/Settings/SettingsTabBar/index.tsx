import React from 'react';
import useStyles from './useStyles';
import {Tabs, Tab} from '@material-ui/core';

const tabs = [
    {
        label: 'general'
    },
    {
        label: 'feedback'
    },
    {
        label: 'follow us'
    },
    {
        label: 'about'
    }
]

const SettingsTabBar = () => {

    const {indicator, tab, customTab} = useStyles()

    const [currentTab, setCurrentTab] = React.useState(0)

    const tabRender = React.useMemo(() => {
        return tabs.map(tab => <Tab className={customTab} key={tab.label} label={tab.label}/>)
    }, [currentTab])


    return (
        <Tabs
            classes={{
            indicator,
        }}
              value={currentTab} onChange={(e, v) => {
            setCurrentTab(v)
        }}>
            {tabRender}
        </Tabs>
    )
};

export default SettingsTabBar;