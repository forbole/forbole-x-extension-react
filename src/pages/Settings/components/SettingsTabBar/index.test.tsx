import React from 'react'
import { fireEvent } from '@testing-library/dom'
import SettingsTabBar from './index'
import { render } from '../../../../jest/test-utils'

describe('components: SettingsTabBar', () => {
  it('renders', () => {
    const t = render(<SettingsTabBar tabs={[{ label: '1' }]} currentTab={0} onChange={jest.fn} />)

    expect(t).toMatchSnapshot()
  })

  it('calls onChange function with correct index when tab is clicked', () => {
    const TEST_FN = jest.fn()
    const TEST_TABS = [{ label: 'test' }, { label: 'tab' }, { label: '123' }]
    const { getByText } = render(
      <SettingsTabBar tabs={TEST_TABS} currentTab={0} onChange={TEST_FN} />
    )

    fireEvent.click(getByText(TEST_TABS[2].label))

    expect(TEST_FN).toHaveBeenCalledWith(2)
  })
})
