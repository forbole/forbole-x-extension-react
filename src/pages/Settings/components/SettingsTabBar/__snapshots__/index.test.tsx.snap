// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components: SettingsTabBar renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiTabs-root css-b7f350-MuiTabs-root"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-9wecxo-MuiButtonBase-root-MuiTab-root"
              role="tab"
              tabindex="0"
              type="button"
            >
              1
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator css-4fp11o-MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiTabs-root css-b7f350-MuiTabs-root"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        style="overflow: hidden; margin-bottom: 0px;"
      >
        <div
          class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-9wecxo-MuiButtonBase-root-MuiTab-root"
            role="tab"
            tabindex="0"
            type="button"
          >
            1
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-4fp11o-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
